### PR TITLE
refactor: remove async, methmeth, propprop

### PR DESCRIPTION
**Pull request by JustinBeckwith**
This moves the system tests more towards async/await, while dropping a few dependencies along the way.

### CODE REVIEW

File: system-test/logging.js [hunk 1]
```diff
@@ -14,20 +14,15 @@
  * limitations under the License.
  */
 
-'use strict';
```
Comment (ofrobots) [inline system-test/logging.js]: Is this intentional? What is the rationale?

Comment (JustinBeckwith) [reply system-test/logging.js]: I am going to run around and do this in all of our TypeScript libs, but we don't need this in our source files.  With the `strict` TypeScript compiler option set, we add `use strict` to all output files by default:
https://github.com/google/ts-style/blob/master/tsconfig-google.json#L14

**Comment by ofrobots** [comment]
Does that apply to js files too?

**Comment by JustinBeckwith** [comment]
Yes! I double checked, and the file in the build folder has `use strict` in it 👍

**Pull request merged**